mmf/models/frcnn.py

```diff
@@ -0,0 +1,555 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+
+"""
+ coding=utf-8
+ Copyright 2018, Antonio Mendoza Hao Tan, Mohit Bansal
+ Adapted From Facebook Inc, Detectron2 && Huggingface Co.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.import copy
+ """
+
+import copy
+import os
+import pickle as pkl
+import shutil
+import tarfile
+from collections import OrderedDict
+from pathlib import Path
+from urllib.parse import urlparse
+from zipfile import ZipFile, is_zipfile
+
+import numpy as np
+import torch
+from filelock import FileLock
+from mmf.utils.configuration import get_mmf_cache_dir
+from torch import nn
+from torch.nn import functional as F
+from torch.nn.modules.batchnorm import BatchNorm2d
+
+
+try:
+    from detectron2.modeling import ShapeSpec
+    from detectron2.modeling.backbone.resnet import BottleneckBlock, ResNet
+    from detectron2.modeling.proposal_generator.rpn import RPN, StandardRPNHead
+    from detectron2.modeling.roi_heads.roi_heads import Res5ROIHeads
+    from detectron2.structures.image_list import ImageList
+except ImportError:
+    pass
+
+
+mmf_cache_home = get_mmf_cache_dir()
```

Comment:
This is not right. We shouldn't call it outside of functions as `cache_dir` for MMF might not be defined by the time you are calling this. This should probably be converted to a function and used to get the value when needed.